apps/api/lib/utils/isValidBase64.ts

```diff
@@ -0,0 +1,17 @@
+export function isValidBase64(input: string): boolean {
+  try {
+    let base64Data = input;
+    // Check if the input starts with a data URI scheme (data:content/type;base64,...)
+    if (input.startsWith("data:")) {
+      const parts = input.split(",");
+      if (parts.length === 2) {
+        // metadata and base64Data
+        base64Data = parts[1]; // Extract the base64 data from the second part
+      }
```

Comment:
```suggestion
      if (parts.length !== 2) return false;
      
      base64Data = parts[1]; // Extract the base64 data from the second part
```

We can return false when parts length !== 2 to not go about further computations as it would be false already.

Reply:
I made this condition to not just cover base64-images, but also to cover any base64 string checks you might have in the app, so the string may be a single part or multiple part based on the usecase.

if we are only going to make this method exclusive to base64-images then there is no need for the above check
and we can tweak the regex you implemented and implement the method as follows:

```export function isValidBase64Image(input: string): boolean {
  const dataURISchemeRegex = /^data:[A-Za-z0-9;:,.=()+\/-]+;base64,/;
  const base64Regex = new RegExp(
    `^(?:${dataURISchemeRegex.source})?(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$`
  );

  return base64Regex.test(input);
}
```
we will also rename the method to be image exclusive `isValidBase64Image`